### PR TITLE
Open the initial conversation when a user first logs in after registering

### PR DESCRIPTION
**Pull request by dalefukami**
### What does this do?

Opens the initial conversation for the user when they've first completed the registration process.

### How do I test this?

Register as a new user and verify that once they've completed the registration process (i.e., set their Name) the app opens and the very first conversation is open and ready.

### CODE REVIEW

File: src/store/channels-list/channels.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { multicastChannel } from 'redux-saga';
+import { call } from 'redux-saga/effects';
+
+let theConversationsChannel;
+export function* conversationsChannel() {
+  if (!theConversationsChannel) {
+    theConversationsChannel = yield call(multicastChannel);
```
Comment (ratik21) [inline src/store/channels-list/channels.ts:7]: what does calling `multicastChannel` do?

File: src/store/registration/saga.ts [hunk 2]
```diff
@@ -140,4 +143,21 @@ export function* saga() {
...
+
+export function* channelsLoaded() {
+  const existingConversationsList = yield select(rawConversationsList());
+  if (existingConversationsList.length > 0) {
+    yield put(setActiveMessengerId(existingConversationsList[0]));
+  }
+}
+
+function* conversationsWatcher() {
+  const channel = yield call(conversationsChannel);
```
Comment (ratik21) [inline src/store/registration/saga.ts:158]: i don't follow this part. This is calling a saga function `conversationsChannel` which is calling a "multicast" channel. Is it fetching/loading our channel(s) info?

File: src/store/registration/saga.ts [hunk 3]
```diff
@@ -140,4 +143,21 @@ export function* saga() {
...
+  const payload = yield take(channel, '*');
+  if (payload.loaded) {
+    yield call(channelsLoaded);
+  }
 }
```
Comment (ratik21) [inline src/store/registration/saga.ts:163]: to clarify, `yield take(channel, '*')` will "listen" for all events were the "channel" redux state is updated, and is checking if `{ loaded: true }` is set? (after which it sets the active messenger id)

**Review by ratik21 — APPROVED** [review]
had a few doubts

File: src/store/channels-list/channels.ts (hunk 1 above)
Comment (dalefukami) [reply src/store/channels-list/channels.ts:7]: `multicastChannel` is a redux-saga thing. It allows you to publish an event that multiple listeners can "take" from.

File: src/store/registration/saga.ts (hunk 2 above)
Comment (dalefukami) [reply src/store/registration/saga.ts:158]: In this line we're just getting access to the channel. The next line "listens" for events on that channel with the `take` call in the same way all the other `take` calls work. When an event comes it'll move on.

File: src/store/registration/saga.ts (hunk 3 above)
Comment (dalefukami) [reply src/store/registration/saga.ts:163]: Not really the "redux state". It's events that come across that specific channel. In this case, it's the conversationsChannel that we're listening to.

The `{loaded: true}` is just the value I chose to use when publishing the event. I decided this conversation event was a "conversations loaded" event and this seemed an ok way to indication that. Another option would be to publish it more like an action payload: `{type: "CONVERSATIONS_LOADED"}`. Maybe that would make more sense?

**Pull request merged**